File: csharp/MaxPainUI/ClientApp/src/app/models/maxpainitem.ts
export class MPChn {
    stock: string;
    stockPrice: number;
    maturity: Date;
    mint: number;
    maxPain: number;
    highCallOI: number;
    highPutOI: number;
    totalCallOI: number;
    totalPutOI: number;
    putCallRatio: number;
    minCash: number;
    maxCash: number;
    createdOn: Date;

    items: Array<MPItem>;

    get totalOI(): number {
        return this.totalCallOI + this.totalPutOI;
    }

    constructor(jsonObj: any) {
        Object.keys(jsonObj).map((key)=>{ this[key] = jsonObj[key] });
        this.items = [];
        for (let i of jsonObj.items) { this.items.push(new MPItem(i)) };
    }
}

export class MPItem {
    s: number;
    coi: number;
    cch: number;
    cpd: number;
    poi: number;
    pch: number;
    ppd: number;
    pd: number;

    get totalOI(): number {
        return this.coi + this.poi;
    }

    get totalCash(): number {
        return this.cch + this.pch;
    }

    constructor(jsonObj: any) {
        Object.keys(jsonObj).map((key)=>{ this[key] = jsonObj[key] });
    }
}

File: csharp/MaxPainInfrastructure/Models/MaxPainItem.cs
namespace MaxPainInfrastructure.Models
{
    public class MPChain
    {
        public string Stock { get; set; }
        public decimal StockPrice { get; set; }
        public DateTime Maturity { get; set; }
        public int Mint { get; set; }

        public decimal MaxPain { get; set; }
        public decimal HighCallOI { get; set; }
        public decimal HighPutOI { get; set; }
        public int TotalCallOI { get; set; }
        public int TotalPutOI { get; set; }
        public decimal PutCallRatio { get; set; }
        public decimal MinCash { get; set; }
        public decimal MaxCash { get; set; }

        public DateTime CreatedOn { get; set; }

        public List<MPItem> Items { get; set; }

        public MPChain()
        {
            Items = new List<MPItem>();
        }
    }

    public class MPItem
    {
        public decimal s { get; set; }

        public int coi { get; set; }
        public decimal cch { get; set; }
        public decimal cpd { get; set; }

        public int poi { get; set; }
        public decimal pch { get; set; }
        public decimal ppd { get; set; }

        public decimal pd { get; set; }

        public decimal TotalCash()
        {
            return cch + pch;
        }
    }
}
